be/src/Project/Services/Catalog/Catalog.Api/Services/Categories/CategoriesService.cs

```diff
@@ -87,62 +87,39 @@ public PagedResults<IEnumerable<CategoryServiceModel>> Get(GetCategoriesServiceM
             };
         }
 
-        public async Task<CategoryServiceModel> GetAsync(GetCategoryServiceModel model)
+        public CategoryServiceModel Get(GetCategoryServiceModel model)
         {
             var categoryItem = _context.Categories.FirstOrDefault(x => x.Id == model.Id && x.IsActive);
 
-            if (categoryItem is not null)
+            if (categoryItem is null)
             {
-                var category = new CategoryServiceModel
-                {
-                    Id = categoryItem.Id,
-                    Order = categoryItem.Order,
-                    Level = categoryItem.Level,
-                    IsLeaf = categoryItem.IsLeaf,
-                    ParentId = categoryItem.Parentid,
-                    LastModifiedDate = categoryItem.LastModifiedDate,
-                    CreatedDate = categoryItem.CreatedDate
-                };
-
-                var thumbnailMedia = _context.CategoryImages.FirstOrDefault(x => x.CategoryId == categoryItem.Id && x.IsActive);
-
-                if (thumbnailMedia is not null)
-                {
-                    category.ThumbnailMediaId = thumbnailMedia.MediaId;
-                }
-
-                var categoryItemTranslations = _context.CategoryTranslations.FirstOrDefault(x => x.Language == model.Language && x.CategoryId == categoryItem.Id && x.IsActive);
-
-                if (categoryItemTranslations is null)
-                {
-                    categoryItemTranslations = _context.CategoryTranslations.FirstOrDefault(x => x.CategoryId == categoryItem.Id && x.IsActive);
-                }
-
-                category.Name = categoryItemTranslations?.Name;
-
-                if (categoryItem.Parentid.HasValue)
-                {
-                    var parentCategoryTranslations = _context.CategoryTranslations.FirstOrDefault(x => x.Language == model.Language && x.CategoryId == categoryItem.Parentid && x.IsActive);
-
-                    if (parentCategoryTranslations is null)
-                    {
-                        parentCategoryTranslations = _context.CategoryTranslations.FirstOrDefault(x => x.CategoryId == categoryItem.Parentid && x.IsActive);
-                    }
+                throw new CustomException(_productLocalizer.GetString("CategoryNotFound"), (int)HttpStatusCode.NoContent);
+            }
 
-                    category.ParentCategoryName = parentCategoryTranslations?.Name;
-                }
+            var translations = _context.CategoryTranslations.Where(x => x.CategoryId == categoryItem.Id && x.IsActive || x.CategoryId == categoryItem.Parentid && x.IsActive).ToList();
 
-                return category;
-            }
+            var thumbnailMedia = _context.CategoryImages.Where(x => x.CategoryId == categoryItem.Id && x.IsActive).ToList();
```

Comment:
**suggestion:** let's take .FirstOrDefault directly since we need only this one in line 110